Particle ray tracing (#1629)
chaos-polymtl/lethe: doc/source/parameters/dem/particle_ray_tracing.rst

=====================
Particle Ray Tracing
=====================

> In this subsection, the parameters are used by the ``lethe-particles-ray-tracing`` application. This application simulates the propagation of photons (or rays) through the simulation domain to reconstruct the surface made by particle using the same principles as profilometry. The parameters defined here control the initial conditions for photon insertion, including their starting positions, directions, and any random offsets applied to their trajectories.

Reviewer: ```suggestion
In this subsection, the parameters used by the ``lethe-particles-ray-tracing`` application are explained. This application simulates the propagation of photons (or rays) through the simulation domain to reconstruct the surface generated by particles using the same principles as profilometry. The parameters control the initial conditions for photon insertion, including their starting positions, directions, and any random offsets applied to their trajectories.
```

> In this subsection, is used when using the ``lethe-particles-ray-tracing`` application. This application simulates the trajectories of photons, or rays, as they propagate through a simulation domain to extract the surface made by particle using the same principal as profilometry. The parameters defined here control the initial conditions for photon insertion, including their starting positions, directions, and any random offsets applied to their trajectories.

Reviewer 2: Not sure to understand the first sentence. Do you mean :

```suggestion
In this subsection, the parameters are used by the ``lethe-particles-ray-tracing`` application. This application simulates the propagation of photons (or rays) through the simulation domain to reconstruct the surface made by particle using the same principles as profilometry. The parameters defined here control the initial conditions for photon insertion, including their starting positions, directions, and any random offsets applied to their trajectories.
```

.. code-block:: text

  subsection particle ray tracing
    set starting photon insertion position                  = 0., 0., 0.
    set insertion unit tensors                              = 1.,0.,0. : 0., 1., 0. : 0., 0., 1.
    set number of inserted photons per direction            = 1 : 1 : 1
    set distance between photons on insertion per direction = 1 : 1 : 1
    set reference displacement vector                       = 0., 0., 1.
    set photon insertion maximum offset                     = 0.
    set photon insertion prn seed                           = 0
    set photon maximum angular offset                       = 0.
    set photon angular offset prn seed                      = 1
  end

-  ``starting photon insertion position`` is the location of the first photon being inserted, given as a 3D coordinate (x,y,z).

> -  ``insertion unit tensors`` are three vectors defining the insertion grid directions relative to the ``starting photon position``. By default, these correspond to the Cartesian axes. The vectors do not need to be orthogonal or normalized, but they must not be zero vectors. The insertion grid is formed by creating insertion point relative to the ``starting photon insertion position`` using these three vectors.

Reviewer: ```suggestion
-  ``insertion unit tensors`` are three vectors defining the insertion grid directions relative to the ``starting photon position``. By default, these correspond to the Cartesian axes (x-y-z). The vectors do not need to be orthogonal or normalized, but they must non-zero vectors. The insertion grid is formed by creating insertion point relative to the ``starting photon insertion position`` using these three vectors.
```
